code/modules/mob/dead/observer/orbit.dm

```diff
@@ -18,94 +27,145 @@
 
 /datum/orbit_menu/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
 	. = ..()
+
 	if(.)
 		return
+
 	switch(action)
-		if ("orbit")
+		if("orbit")
 			var/ref = params["ref"]
-			var/atom/movable/poi = (locate(ref) in GLOB.mob_list) || (locate(ref) in GLOB.poi_list)
-			if (poi == null)
-				. = TRUE
-				return
-			owner.ManualFollow(poi)
-			owner.reset_perspective(null)
+			var/auto_observe = params["auto_observe"]
+			var/atom/poi = SSpoints_of_interest.get_poi_atom_by_ref(ref)
+
+			if((ismob(poi) && !SSpoints_of_interest.is_valid_poi(poi, CALLBACK(src, .proc/validate_mob_poi))) \
+				|| !SSpoints_of_interest.is_valid_poi(poi))
```

Comment:
```suggestion
				|| !SSpoints_of_interest.is_valid_poi(poi) \
			)
```

Minor nitpick, you don't have to, but it makes it clearer where the condition ends

Reply:
Minor nitpicks are our bread and butter.